core/src/main/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializer.scala

```diff
@@ -26,7 +27,7 @@ import org.apache.spark.status.KVUtils.KVStoreScalaSerializer
 private[spark] class KVStoreProtobufSerializer extends KVStoreScalaSerializer {
   override def serialize(o: Object): Array[Byte] =
     KVStoreProtobufSerializer.getSerializer(o.getClass) match {
-      case Some(serializer) => serializer.serialize(o)
+      case Some(serializer) => serializer.serialize(o.asInstanceOf[Any])
```

Comment:
Do we really need call `asInstanceOf[Any]`?

Reply:
shouldn't be necessary, remove it in [2568fc0](https://github.com/apache/spark/pull/39487/commits/2568fc0d654ed13c04bd335ece01503d3d5e0c71), waiting CI